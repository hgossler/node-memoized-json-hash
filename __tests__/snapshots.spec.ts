/**
 * These snapshot tests are used to detect whether the generated hashes are consistent.
 * If not, then this should result in a major version bump.
 */
import jsonHash from '../index';
import id from 'object-hash';


const hex = {
	md5: /^[0-9a-f]{32}$/,
	sha1: /^[0-9a-f]{40}$/,
	sha256: /^[0-9a-f]{64}$/,
};

describe('Snapshots', () => {
	const numbers = [
		10,
		1.1,
		NaN,
		Infinity
	];

	const primitives = [
		...numbers,
		true,
		false,
		null,
		'a string'
	];

	const objects = [
		{},
		{foo: 'bar', bar: 'baz'},
		{foo: {bar: true, bax: 1}},
		{foo: ['bar', 'baz'], bax: true},
		['foo', 'bar'],
		{defined: true, not_defined: undefined},
		{a: undefined, b: 1, c: undefined},
		{'!@"#$/\'``': 'unusual...'}
	];

	const specials = [
		{map: new Map<any,any>([[1, 2], [{}, []]])},
		{set: new Set([1,2, {}, [], new Map()])},
		[true, undefined]
	];


	const all = [
		...primitives,
		...objects,
		...specials
	];


	for (let item of all) {
		test('hash snapshot (default options)', () => {
			expect(jsonHash(item)).toMatchSnapshot(id(item));
		});


		test('hash snapshot (algorithm: none)', () => {
			const string = jsonHash(item, {algorithm: 'none'});

			expect(string).toMatchSnapshot(id(item));
		});


		test('hash snapshot (algorithm: md5)', () => {
			const string = jsonHash(item, {algorithm: 'md5'});

			expect(string).toMatch(hex.md5);
			expect(string).toMatchSnapshot(id(item));
		});


		test('hash snapshot (algorithm: sha1)', () => {
			const string = jsonHash(item, {algorithm: 'sha1'});

			expect(string).toMatch(hex.sha1);
			expect(string).toMatchSnapshot(id(item));
		});


		test('hash snapshot (algorithm: sha256)', () => {
			const string = jsonHash(item, {algorithm: 'sha256'});

			expect(string).toMatch(hex.sha256);
			expect(string).toMatchSnapshot(id(item));
		});


		test('produces valid json (algorithm: none)', () => {
			const string = jsonHash(item, {algorithm: 'none'});

			expect(() => JSON.parse(string)).not.toThrowError();
		});
	}


	for(let item of objects) {
		test( 'parsing stringified json yields original object (algorithm: none)', () => {
			const string = jsonHash(item, {algorithm: 'none'});
			const parsed = JSON.parse(string);

			expect(parsed).toEqual(item);
		});
	}
});