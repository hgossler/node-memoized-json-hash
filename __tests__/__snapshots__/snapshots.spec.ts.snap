// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots hash snapshot (algorithm: md5): 0869d0538a8bc81de4d0672fed972c5edb1df9da 1`] = `"78d1ce3583498a65122be5748418fbf9"`;

exports[`Snapshots hash snapshot (algorithm: md5): 0a829f62a5cbf3a7e5d452cfc45351e4804f34af 1`] = `"976522e1afb4b5a3a79654ac950e7a31"`;

exports[`Snapshots hash snapshot (algorithm: md5): 0b347dd5c9da9a76a0efb0e3b47c5d732ca5da71 1`] = `"eef6521b6a3dc05a8f797e55db85db5d"`;

exports[`Snapshots hash snapshot (algorithm: md5): 1e1360e9e74f24ff2f26c8d930fc37255259ebd2 1`] = `"777d45bbbcdf50d49c42c70ad7acf5fe"`;

exports[`Snapshots hash snapshot (algorithm: md5): 1f1cd499866381b5699f050e1f1bd3ef06a5be49 1`] = `"60dc2cb9520db75fb07f53e562521e1a"`;

exports[`Snapshots hash snapshot (algorithm: md5): 2b3d64c68adcdad2c3b850deb03280b5746828da 1`] = `"37a6259cc0c1dae299a7866489dff0bd"`;

exports[`Snapshots hash snapshot (algorithm: md5): 3c38adefd2f5bf45341cc50364d58a07feb42907 1`] = `"37a6259cc0c1dae299a7866489dff0bd"`;

exports[`Snapshots hash snapshot (algorithm: md5): 5c34e1fe67e9d6c8ba988f2aaa95b0fe1f3ad8e5 1`] = `"9c0ff000cafcf29a855f1aea5fb2d1f0"`;

exports[`Snapshots hash snapshot (algorithm: md5): 8cf9ece02759c073df613079223901ace2d0df91 1`] = `"42e54c254c27b98fed6e65a788eac4c7"`;

exports[`Snapshots hash snapshot (algorithm: md5): 63d0b026e716c2f15878d2b9d7ff673afe0f9916 1`] = `"d3d9446802a44259755d38e6d163e820"`;

exports[`Snapshots hash snapshot (algorithm: md5): 7980d15b7dde9130c38473c1a96760c396ae5820 1`] = `"1aae9c14bd1af0f06ab0cb0320b477db"`;

exports[`Snapshots hash snapshot (algorithm: md5): 9314f52c3fac509ec9b133647ceca91731d07db4 1`] = `"e17d0ee99013e5ee01d0b073ed2bab2c"`;

exports[`Snapshots hash snapshot (algorithm: md5): 109085beaaa80ac89858b283a64f7c75d7e5bb12 1`] = `"37a6259cc0c1dae299a7866489dff0bd"`;

exports[`Snapshots hash snapshot (algorithm: md5): 323217f643c3e3f1fe7532e72ac01bb0748c97be 1`] = `"99914b932bd37a50b983c5e7c90ae93b"`;

exports[`Snapshots hash snapshot (algorithm: md5): a1f3c9dd19f849b024779d85fcc8d882c7a9d369 1`] = `"1ea13cb52ddd7c90e9f428d1df115d8f"`;

exports[`Snapshots hash snapshot (algorithm: md5): b29c63990dea846689120516761de20c056e3539 1`] = `"68934a3e9455fa72420237eb05902327"`;

exports[`Snapshots hash snapshot (algorithm: md5): cdf22d2a18b96ef07f6105cd8093ae12a8772cb3 1`] = `"b326b5062b2f0e69046810717534cb09"`;

exports[`Snapshots hash snapshot (algorithm: md5): d4df21fea9151b8d1df8c2f6217ed26ba334859e 1`] = `"6e224989555a8dcd3af6ff5afb357de2"`;

exports[`Snapshots hash snapshot (algorithm: md5): f5295a9aab1ef81a413a8963b79e20eb52225f31 1`] = `"766cc9e94939b19ccb625a8b60e95824"`;

exports[`Snapshots hash snapshot (algorithm: none): 0869d0538a8bc81de4d0672fed972c5edb1df9da 1`] = `"{\\"bax\\":true,\\"foo\\":[\\"bar\\",\\"baz\\"]}"`;

exports[`Snapshots hash snapshot (algorithm: none): 0a829f62a5cbf3a7e5d452cfc45351e4804f34af 1`] = `"{\\"map\\":{}}"`;

exports[`Snapshots hash snapshot (algorithm: none): 0b347dd5c9da9a76a0efb0e3b47c5d732ca5da71 1`] = `"\\"a string\\""`;

exports[`Snapshots hash snapshot (algorithm: none): 1e1360e9e74f24ff2f26c8d930fc37255259ebd2 1`] = `"1.1"`;

exports[`Snapshots hash snapshot (algorithm: none): 1f1cd499866381b5699f050e1f1bd3ef06a5be49 1`] = `"{\\"bar\\":\\"baz\\",\\"foo\\":\\"bar\\"}"`;

exports[`Snapshots hash snapshot (algorithm: none): 2b3d64c68adcdad2c3b850deb03280b5746828da 1`] = `"null"`;

exports[`Snapshots hash snapshot (algorithm: none): 3c38adefd2f5bf45341cc50364d58a07feb42907 1`] = `"null"`;

exports[`Snapshots hash snapshot (algorithm: none): 5c34e1fe67e9d6c8ba988f2aaa95b0fe1f3ad8e5 1`] = `"{\\"defined\\":true}"`;

exports[`Snapshots hash snapshot (algorithm: none): 8cf9ece02759c073df613079223901ace2d0df91 1`] = `"{\\"set\\":{}}"`;

exports[`Snapshots hash snapshot (algorithm: none): 63d0b026e716c2f15878d2b9d7ff673afe0f9916 1`] = `"10"`;

exports[`Snapshots hash snapshot (algorithm: none): 7980d15b7dde9130c38473c1a96760c396ae5820 1`] = `"{\\"foo\\":{\\"bar\\":true,\\"bax\\":1}}"`;

exports[`Snapshots hash snapshot (algorithm: none): 9314f52c3fac509ec9b133647ceca91731d07db4 1`] = `"{\\"b\\":1}"`;

exports[`Snapshots hash snapshot (algorithm: none): 109085beaaa80ac89858b283a64f7c75d7e5bb12 1`] = `"null"`;

exports[`Snapshots hash snapshot (algorithm: none): 323217f643c3e3f1fe7532e72ac01bb0748c97be 1`] = `"{}"`;

exports[`Snapshots hash snapshot (algorithm: none): a1f3c9dd19f849b024779d85fcc8d882c7a9d369 1`] = `"[\\"foo\\",\\"bar\\"]"`;

exports[`Snapshots hash snapshot (algorithm: none): b29c63990dea846689120516761de20c056e3539 1`] = `"false"`;

exports[`Snapshots hash snapshot (algorithm: none): cdf22d2a18b96ef07f6105cd8093ae12a8772cb3 1`] = `"true"`;

exports[`Snapshots hash snapshot (algorithm: none): d4df21fea9151b8d1df8c2f6217ed26ba334859e 1`] = `"{\\"!@\\\\\\"#$/'\`\`\\":\\"unusual...\\"}"`;

exports[`Snapshots hash snapshot (algorithm: none): f5295a9aab1ef81a413a8963b79e20eb52225f31 1`] = `"[true,null]"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 0869d0538a8bc81de4d0672fed972c5edb1df9da 1`] = `"b52697b879cb32a679dbcd7a3591e453de5bbd58"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 0a829f62a5cbf3a7e5d452cfc45351e4804f34af 1`] = `"34c5651c81c5421bafa6019977dff2ccb3432f02"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 0b347dd5c9da9a76a0efb0e3b47c5d732ca5da71 1`] = `"3ba59f8353a8c3a833783e85dee4cfacced42d5a"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 1e1360e9e74f24ff2f26c8d930fc37255259ebd2 1`] = `"4491f8b5ecc09d9de308ea37e6a23ad2dd66c256"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 1f1cd499866381b5699f050e1f1bd3ef06a5be49 1`] = `"7337b77ab6e6c7eb8e0ccd547591f6b926cec1e6"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 2b3d64c68adcdad2c3b850deb03280b5746828da 1`] = `"2be88ca4242c76e8253ac62474851065032d6833"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 3c38adefd2f5bf45341cc50364d58a07feb42907 1`] = `"2be88ca4242c76e8253ac62474851065032d6833"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 5c34e1fe67e9d6c8ba988f2aaa95b0fe1f3ad8e5 1`] = `"60eeef7a520b2e32d219b513868053c5b484e5e3"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 8cf9ece02759c073df613079223901ace2d0df91 1`] = `"66d4c625d9fb5a421a48574a455fb08a2fdff746"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 63d0b026e716c2f15878d2b9d7ff673afe0f9916 1`] = `"b1d5781111d84f7b3fe45a0852e59758cd7a87e5"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 7980d15b7dde9130c38473c1a96760c396ae5820 1`] = `"13cb01f8b37b2317904a991cfbb33c5b37e2427e"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 9314f52c3fac509ec9b133647ceca91731d07db4 1`] = `"ee6093f27ea000b9b837f4a91ac7199131483d31"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 109085beaaa80ac89858b283a64f7c75d7e5bb12 1`] = `"2be88ca4242c76e8253ac62474851065032d6833"`;

exports[`Snapshots hash snapshot (algorithm: sha1): 323217f643c3e3f1fe7532e72ac01bb0748c97be 1`] = `"bf21a9e8fbc5a3846fb05b4fa0859e0917b2202f"`;

exports[`Snapshots hash snapshot (algorithm: sha1): a1f3c9dd19f849b024779d85fcc8d882c7a9d369 1`] = `"dedf54a84ba7aaa921dda4e0f46edd5f07f7639c"`;

exports[`Snapshots hash snapshot (algorithm: sha1): b29c63990dea846689120516761de20c056e3539 1`] = `"7cb6efb98ba5972a9b5090dc2e517fe14d12cb04"`;

exports[`Snapshots hash snapshot (algorithm: sha1): cdf22d2a18b96ef07f6105cd8093ae12a8772cb3 1`] = `"5ffe533b830f08a0326348a9160afafc8ada44db"`;

exports[`Snapshots hash snapshot (algorithm: sha1): d4df21fea9151b8d1df8c2f6217ed26ba334859e 1`] = `"f95dfa14d12b44b8194af6189b832e5b5b6e0156"`;

exports[`Snapshots hash snapshot (algorithm: sha1): f5295a9aab1ef81a413a8963b79e20eb52225f31 1`] = `"984aaefa1c08852d73eca6bf5f6c321c6c28d92a"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 0869d0538a8bc81de4d0672fed972c5edb1df9da 1`] = `"83a15e0190334a91e274baf0f90602f29276699a0429d992ca29b7e712a6e9fc"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 0a829f62a5cbf3a7e5d452cfc45351e4804f34af 1`] = `"65061da03da794ff0b85ed3b8e7d9324e8d962935307977d6003c043f4e57014"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 0b347dd5c9da9a76a0efb0e3b47c5d732ca5da71 1`] = `"c95d6352f524a770a787c16509237baf7eb59967699fb9a6d825270e7ec0eacf"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 1e1360e9e74f24ff2f26c8d930fc37255259ebd2 1`] = `"b05e244762b1e472be89a93800cc3ee326743cecb55984bf12813addb8de66d0"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 1f1cd499866381b5699f050e1f1bd3ef06a5be49 1`] = `"ad378c8d2a976d71cf5eed1717ff09426e378da99d7257547841a5a04343362d"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 2b3d64c68adcdad2c3b850deb03280b5746828da 1`] = `"74234e98afe7498fb5daf1f36ac2d78acc339464f950703b8c019892f982b90b"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 3c38adefd2f5bf45341cc50364d58a07feb42907 1`] = `"74234e98afe7498fb5daf1f36ac2d78acc339464f950703b8c019892f982b90b"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 5c34e1fe67e9d6c8ba988f2aaa95b0fe1f3ad8e5 1`] = `"2f8b66c2f01186bbf042d4d9a9708651bd2432b2c24454b1e14bec00ce3cb55f"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 8cf9ece02759c073df613079223901ace2d0df91 1`] = `"4fbe5b65b5a07e0d6aa3e8b01098b3d43deddedfb61d4dae6ff1130dec5c4be3"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 63d0b026e716c2f15878d2b9d7ff673afe0f9916 1`] = `"4a44dc15364204a80fe80e9039455cc1608281820fe2b24f1e5233ade6af1dd5"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 7980d15b7dde9130c38473c1a96760c396ae5820 1`] = `"39aa192ca223d86eb95c6a4c94d0ebed6be84c6654584032c49e5f47ccdff00f"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 9314f52c3fac509ec9b133647ceca91731d07db4 1`] = `"eb8ed3ccb5023093b56f490a46501e88d09736687e609fdbc1c71b3df8b9ccd3"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 109085beaaa80ac89858b283a64f7c75d7e5bb12 1`] = `"74234e98afe7498fb5daf1f36ac2d78acc339464f950703b8c019892f982b90b"`;

exports[`Snapshots hash snapshot (algorithm: sha256): 323217f643c3e3f1fe7532e72ac01bb0748c97be 1`] = `"44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a"`;

exports[`Snapshots hash snapshot (algorithm: sha256): a1f3c9dd19f849b024779d85fcc8d882c7a9d369 1`] = `"a9827ca249365f016ed0a9da9604634799792532c0559be0547348d44bbddbde"`;

exports[`Snapshots hash snapshot (algorithm: sha256): b29c63990dea846689120516761de20c056e3539 1`] = `"fcbcf165908dd18a9e49f7ff27810176db8e9f63b4352213741664245224f8aa"`;

exports[`Snapshots hash snapshot (algorithm: sha256): cdf22d2a18b96ef07f6105cd8093ae12a8772cb3 1`] = `"b5bea41b6c623f7c09f1bf24dcae58ebab3c0cdd90ad966bc43a45b44867e12b"`;

exports[`Snapshots hash snapshot (algorithm: sha256): d4df21fea9151b8d1df8c2f6217ed26ba334859e 1`] = `"c112e3f9b682a85986ddf653b5f2abcdd617f178ef145f0ef048f6fda48ecb7a"`;

exports[`Snapshots hash snapshot (algorithm: sha256): f5295a9aab1ef81a413a8963b79e20eb52225f31 1`] = `"d94613877de59a3a0b3717d90975186f4eec3b2ea86e58be2ae86012529b845d"`;

exports[`Snapshots hash snapshot (default options): 0869d0538a8bc81de4d0672fed972c5edb1df9da 1`] = `"b52697b879cb32a679dbcd7a3591e453de5bbd58"`;

exports[`Snapshots hash snapshot (default options): 0a829f62a5cbf3a7e5d452cfc45351e4804f34af 1`] = `"34c5651c81c5421bafa6019977dff2ccb3432f02"`;

exports[`Snapshots hash snapshot (default options): 0b347dd5c9da9a76a0efb0e3b47c5d732ca5da71 1`] = `"3ba59f8353a8c3a833783e85dee4cfacced42d5a"`;

exports[`Snapshots hash snapshot (default options): 1e1360e9e74f24ff2f26c8d930fc37255259ebd2 1`] = `"4491f8b5ecc09d9de308ea37e6a23ad2dd66c256"`;

exports[`Snapshots hash snapshot (default options): 1f1cd499866381b5699f050e1f1bd3ef06a5be49 1`] = `"7337b77ab6e6c7eb8e0ccd547591f6b926cec1e6"`;

exports[`Snapshots hash snapshot (default options): 2b3d64c68adcdad2c3b850deb03280b5746828da 1`] = `"2be88ca4242c76e8253ac62474851065032d6833"`;

exports[`Snapshots hash snapshot (default options): 3c38adefd2f5bf45341cc50364d58a07feb42907 1`] = `"2be88ca4242c76e8253ac62474851065032d6833"`;

exports[`Snapshots hash snapshot (default options): 5c34e1fe67e9d6c8ba988f2aaa95b0fe1f3ad8e5 1`] = `"60eeef7a520b2e32d219b513868053c5b484e5e3"`;

exports[`Snapshots hash snapshot (default options): 8cf9ece02759c073df613079223901ace2d0df91 1`] = `"66d4c625d9fb5a421a48574a455fb08a2fdff746"`;

exports[`Snapshots hash snapshot (default options): 63d0b026e716c2f15878d2b9d7ff673afe0f9916 1`] = `"b1d5781111d84f7b3fe45a0852e59758cd7a87e5"`;

exports[`Snapshots hash snapshot (default options): 7980d15b7dde9130c38473c1a96760c396ae5820 1`] = `"13cb01f8b37b2317904a991cfbb33c5b37e2427e"`;

exports[`Snapshots hash snapshot (default options): 9314f52c3fac509ec9b133647ceca91731d07db4 1`] = `"ee6093f27ea000b9b837f4a91ac7199131483d31"`;

exports[`Snapshots hash snapshot (default options): 109085beaaa80ac89858b283a64f7c75d7e5bb12 1`] = `"2be88ca4242c76e8253ac62474851065032d6833"`;

exports[`Snapshots hash snapshot (default options): 323217f643c3e3f1fe7532e72ac01bb0748c97be 1`] = `"bf21a9e8fbc5a3846fb05b4fa0859e0917b2202f"`;

exports[`Snapshots hash snapshot (default options): a1f3c9dd19f849b024779d85fcc8d882c7a9d369 1`] = `"dedf54a84ba7aaa921dda4e0f46edd5f07f7639c"`;

exports[`Snapshots hash snapshot (default options): b29c63990dea846689120516761de20c056e3539 1`] = `"7cb6efb98ba5972a9b5090dc2e517fe14d12cb04"`;

exports[`Snapshots hash snapshot (default options): cdf22d2a18b96ef07f6105cd8093ae12a8772cb3 1`] = `"5ffe533b830f08a0326348a9160afafc8ada44db"`;

exports[`Snapshots hash snapshot (default options): d4df21fea9151b8d1df8c2f6217ed26ba334859e 1`] = `"f95dfa14d12b44b8194af6189b832e5b5b6e0156"`;

exports[`Snapshots hash snapshot (default options): f5295a9aab1ef81a413a8963b79e20eb52225f31 1`] = `"984aaefa1c08852d73eca6bf5f6c321c6c28d92a"`;
